src/main/java/com/redhat/cloud/custompolicies/app/model/Policy.java

```diff
@@ -44,6 +44,23 @@
   public String conditions;
   public String actions;
 
+  // De-serialise the array form of actions coming from the UI
+  // this needs more work with some more typesafe translations
+  // or alternatively storing of actions as json blob
+  // See CPOL-33
+  public void setActions(List<Map<String, String>> actionList) {
+    System.out.println(actionList);
+    Iterator<Map<String,String>> ite = actionList.iterator();
+    actions = "";
```

Comment:
Wouldn't StringBuilder make more sense here? Otherwise, the code would be quite a bit more simple I guess with something like:

```java
actionList.stream().map(m -> m.get("type")).collect(Collectors.join(", "));
```

Or 

```
StringJoiner joiner = new StringJoiner(", ");
for(Map<String, String map> map : actionList) {
joiner.add(map.get("type"));
}
```

Just all the tricks in the Java 8+ world ;)

Reply:
Yes it would. I see this code as very temporary -- see comment about CPOL-33.
But I can adopt your suggestions anyway